pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -451,6 +451,10 @@ func buildMgmtPortListeners(managementPorts model.PortList, managementIP string)
 // buildInboundAuth adds TLS to the listener if the policy requires one.
 func buildSidecarListenerTLSContext(authenticationPolicy *authn.Policy) *auth.DownstreamTlsContext {
 	if requireTLS, mTLSParams := model.RequireTLS(authenticationPolicy); requireTLS {
+		if mTLSParams == nil {
```

Comment:
This case cannot occur given the current implementation and contract of the method. This check is not needed, just remove.

https://github.com/istio/istio/blob/654617da2c7f12a6e5fc97a1f2779f1cae23cb4f/pilot/pkg/model/authentication.go#L86-L103

Reply:
Both cases were added because tests showed it to happen, and the result is the entire config is rejected.

Removed the extra checks to get the PR in - but I don't agree that validation should only be done in some layer and not in another. Envoy is re-validating it again - and it's quire reasonable to have have 'validation in depth'.